### ISSUE TITLE
loss cls and loss total does not  decrease

### ISSUE THREAD

**Issue by sadransh** [zylo117/Yet-Another-EfficientDet-Pytorch#495]
I am experiencing a problem with training head only with the following command : 
`python train.py -c 4 -p crack --batch_size 30 --lr 1e-1 --num_epochs 100  --load_weights efficientdet-d4.pth  --head_only True --optim sgd` 

used with/without --optim , used different batch sizes and lr s and also 2 d s . 
```

categories : 
    "categories": [
        {
            "supercategory": "none",
            "id": 1,
            "name": "cat1"
        },
        {
            "supercategory": "none",
            "id": 2,
            "name": "cat2"
        },
...
    ]

```
this is loss after this many epochs.

```
Step: 785. Epoch: 1/100. Iteration: 155/631. Cls loss: 146298.14062. Reg loss: 2.82531. Total loss: 146300.96875:  24%|▏| 1

Step: 1549. Epoch: 2/100. Iteration: 288/631. Cls loss: 196628.09375. Reg loss: 2.85675. Total loss: 196630.95312:  46%|▍|
```

I have already used my coco set with another repo(detectron2) and got acceptable results. So the problem should not be because of annotations.

my image sizes are around 900by900 and around 12K images.

**Comment by WangBoying** [comment]
I have met the same problem!

**Comment by WangBoying** [comment]
python  train.py -c 0 -p coco2017 --batch_size 32 --lr 4e-4 --load_weights weights/efficientdet-d0.pth --head_only True
Step: 59999. Epoch: 65/500. Iteration: 200/920. Cls loss: 0.49584. Reg loss: 2.83787. Total loss: 3.33370:  22%|██▏       | 199/920 [01:13<04:22,  2.75it/s]checkpoint...
Step: 60499. Epoch: 65/500. Iteration: 700/920. Cls loss: 0.49851. Reg loss: 2.58777. Total loss: 3.08627:  76%|███████▌  | 699/920 [04:03<01:30,  2.43it/s]checkpoint...
Step: 60719. Epoch: 65/500. Iteration: 920/920. Cls loss: 0.54958. Reg loss: 2.77278. Total loss: 3.32237: 100%|██████████| 920/920 [05:20<00:00,  2.87it/s]
Val. Epoch: 65/500. Classification loss: 0.77456. Regression loss: 3.86610. Total loss: 4.64066

**Comment by alexriedel1** [comment]
Both of you should go on with the training for a while, i guess it will decrease later

**Comment by LoveIsAGame** [comment]
i have met the same problem ,and i  trained a lot of iterations, and the situation still didn't improve, especially reg_loss. I wonder if this problem has been solved？

**Comment by alexriedel1** [comment]
how many steps/epochs?

**Comment by LoveIsAGame** [comment]
![图片](https://user-images.githubusercontent.com/49852180/95573660-5db0bf00-0a5e-11eb-9b6a-d2e13367baf5.png)

**Comment by alexriedel1** [comment]
Sure, 13,000 Iterations is nothing, for a custom dataset i'm getting losses below 1 after maybe 40,000 Iterations

**Comment by LoveIsAGame** [comment]
The main reason is that reg stopped decreasing a long time ago, about the 1000th epoch

**Comment by alexriedel1** [comment]
Sorry I dont get this. Did you train for 500 or for 100 Epochs? Whats your reg loss? maybe can you provide some more information?

**Comment by LoveIsAGame** [comment]
Sorry, I didn't provide enough information. I'll retrain my dataset and give you detailed results tomorrow. Hope you can find time to reply me tomorrow, thank you!

**Comment by alexriedel1** [comment]
Sure!

**Comment by LoveIsAGame** [comment]
![图片](https://user-images.githubusercontent.com/49852180/95575598-8e462800-0a61-11eb-8be7-65808034543a.png)
Is this OK? and epoch is enouch?

**Comment by alexriedel1** [comment]
Yeah give it a try. Maybe you can also show a tensorboard graph of the losses over iterations after your training :)

**Comment by LoveIsAGame** [comment]
![图片](https://user-images.githubusercontent.com/49852180/95579133-a620aa80-0a67-11eb-8ab1-91d4cc0a2f66.png)
I don't know if this interruption is normal during training

**Comment by alexriedel1** [comment]
well as you started training with evaluation every 100 epochs, this seems pretty normal to me

**Comment by LoveIsAGame** [comment]
I don't know exactly what step means here

**Comment by LoveIsAGame** [comment]
![图片](https://user-images.githubusercontent.com/49852180/95643308-01db4a00-0ae1-11eb-8f53-2f1d3740717b.png)


I accidentally ended the training. What order should I use to continue training？

**Comment by alexriedel1** [comment]
Please check the documentation, its under https://github.com/zylo117/Yet-Another-EfficientDet-Pytorch#5-resume-training

but your loss is starting to look good right?

**Comment by LoveIsAGame** [comment]
but your loss is starting to look good right? yes! but I trained again yesterday！
![图片](https://user-images.githubusercontent.com/49852180/95669805-125df400-0bb7-11eb-8e4d-525d44b65791.png)
It's the command I entered！
![图片](https://user-images.githubusercontent.com/49852180/95669831-5e109d80-0bb7-11eb-90a9-ff6b5e168502.png)
![图片](https://user-images.githubusercontent.com/49852180/95669840-71bc0400-0bb7-11eb-9476-2c8b5fddb61c.png)


This is the current training situation, not good at val. And loss seems to be stabilizing. Do I continue training now? Can you give me some advice?

**Comment by LoveIsAGame** [comment]
Exactly the same training, I don't know why the effect is so bad！

**Comment by zylo117** [comment]
The loss should be under 0.5 but yours seems to be underfitting

**Comment by zylo117** [comment]
You can try running the tutorial on shape dataset to get a better overview.
